src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/gui/CustomCubicGui.java

```diff
@@ -252,7 +259,9 @@ public void onClick(UIButton.ClickEvent evt) {
     }
 
     private void done() {
-        parent.chunkProviderSettingsJson = getSettingsJson(getConfig(), true);
+        getConfig().save(new File(Minecraft.getMinecraft().mcDataDir, "saves/"+parent.saveDirName+"/"));
+        String settingsJsonString = getSettingsJson(getConfig(), true);
+        parent.chunkProviderSettingsJson = settingsJsonString;
```

Comment:
Saving should be done when saving the world, once the world already exists. This may need some mixins, specifically into `GuiCreateWorld`. And you should NOT set the generator setting string as is into vanilla. This WILL break if it's too long. That's the entire reason it needs to be stored externally in the first place.